administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
+        {
+          $new_width = $srcWidth;
+        }
+        if($srcHeight < $new_height)
+        {
+          $new_height = $srcHeight;
+        }
+        // Expand the thumbnail's aspect ratio
+        // to fit the width/height of the image
+        $ratiowidth = $srcWidth / $new_width;
+        $ratioheight = $srcHeight / $new_height;
+        if ($ratiowidth < $ratioheight)
+        {
+          $ratio = $ratiowidth;
+        }
+        else
+        {
+          $ratio = $ratioheight;
+        }
+        // Calculate the offsets for cropping the source image according
+        // to thumbposition
+        switch($cropposition)
+        {
+          // Left upper corner
+          case 0:
+            $offsetx = 0;
+            $offsety = 0;
+            break;
+          // Right upper corner
+          case 1:
+            $offsetx = floor(($srcWidth - ($new_width * $ratio)));
+            $offsety = 0;
+            break;
+          // Left lower corner
+          case 3:
+            $offsetx = 0;
+            $offsety = floor(($srcHeight - ($new_height * $ratio)));
+            break;
+          // Right lower corner
+          case 4:
+            $offsetx = floor(($srcWidth - ($new_width * $ratio)));
+            $offsety = floor(($srcHeight - ($new_height * $ratio)));
+            break;
+          // Default center
+          default:
+            $offsetx = floor(($srcWidth - ($new_width * $ratio)) * 0.5);
+            $offsety = floor(($srcHeight - ($new_height * $ratio)) * 0.5);
+            break;
+        }
       }
-    }
-
-    if(is_null($offsetx) && is_null($offsety))
-    {
-      $ratio = max($ratio, 1.0);
+      if(is_null($offsetx) && is_null($offsety))
+      {
+        $ratio = max($ratio, 1.0);
 
-      $destWidth  = (int)($srcWidth / $ratio);
-      $destHeight = (int)($srcHeight / $ratio);
-    }
-    else
-    {
-      $destWidth = $new_width;
-      $destHeight = $thumbheight;
-      $srcWidth  = (int)($destWidth * $ratio);
-      $srcHeight = (int)($destHeight * $ratio);
-    }
+        $destWidth  = (int)($srcWidth / $ratio);
+        $destHeight = (int)($srcHeight / $ratio);
+      }
+      else
+      {
+        $destWidth = $new_width;
+        $destHeight = $new_height;
+        $srcWidth  = (int)($destWidth * $ratio);
+        $srcHeight = (int)($destHeight * $ratio);
+      }
 
     // Method for creation of the resized image
     switch($method)
     {
       case 'gd1':
+        $debugoutput.='GD1...<br/>';
+        // no animated gif support
         if(!function_exists('imagecreatefromjpeg'))
+        // check, if GD is available
         {
           $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_LIBARY_NOT_INSTALLED');
           return false;
         }
-        if($imginfo[2] == 'JPG')
+        // create empty image of specified size
+        $dst_frames = array(imagecreate($destWidth, $destHeight));
+        $src_frames = JoomFile::imageCreateFrom_GD($src_file, $dst_frames, $imginfo);
```

Comment:
imageCreateFrom_GD: external function which serves as a replacement for imagecreatefromjpg. It creates a GD object from an existing image (from path) of a specific file-type